socorro/unittest/processor/rules/test_mozilla.py

```diff
@@ -1384,90 +1384,100 @@ def test_aurora_channel(self):
         assert processor_meta.processor_notes == []
 
 
-class TestOsPrettyName(object):
-    def test_everything_we_hoped_for(self):
+class TestOsPrettyName:
+    @pytest.mark.parametrize(
+        "os_name, os_version, expected",
+        [
+            # Known windows version
+            ("Windows NT", "10.0.11.7600", "Windows 10"),
+            # Unknown windows version
+            ("Windows NT", "15.2", "Windows Unknown"),
+            # A valid version of Mac OS X
+            ("Mac OS X", "10.18.324", "OS X 10.18"),
+            # An invalid version of Mac OS X
+            ("Mac OS X", "12.1", "OS X Unknown"),
+            # Generic Linux
+            ("Linux", "0.0.12.13", "Linux"),
+        ],
+    )
+    def test_everything_we_hoped_for(self, os_name, os_version, expected):
```

Comment:
While working on this, I redid the tests so they're separate parametrized tests rather than a single test testing a bazillion different things.